### ISSUE TITLE
INSERT JSON: if PK value is null, received not clear error message, different from Cassandra message

### ISSUE THREAD

**Issue by juliayakovlev** [scylladb/scylla#3665]
Scylla version (or git commit hash): master, last commit 6b179b0

Scenario:
1. CREATE TABLE primitive_type_test (key1 text PRIMARY KEY, col1 ascii)
2. INSERT INTO primitive_type_test (col1) values ('bar')

Scylla error message:
_ServerError: Attempt to access value of a disengaged optional object_

Cassandra error message:
_InvalidRequest: code=2200 [Invalid query] message="Invalid null value in condition for column key1"_

**Comment by tzach** [comment]
```
CREATE TABLE primitive_type_test (key1 text PRIMARY KEY, col1 ascii)
INSERT INTO primitive_type_test (col1) values ('bar')
```

@juliayakovlev is this the right example?
You should not be able to insert a row without the primary key, key1 in this case.

**Comment by duarten** [comment]
@tzach The issue is that we fail internally instead of sending an error back to the user.

**Comment by psarna** [comment]
Actually, `INSERT JSON` statement is not used anywhere in the issue description, so I think the issue topic should be changed. And on 55d2fdee9a3e303ce08f66a564a3bf2d10e3fde4 master it returns:
~~~
CREATE TABLE primitive_type_test (key1 text PRIMARY KEY, col1 ascii);
INSERT INTO primitive_type_test (col1) values ('bar');
InvalidRequest: Error from server: code=2200 [Invalid query] message="Missing mandatory PRIMARY KEY part key1"
~~~
which sounds right.

**Comment by duarten** [comment]
That makes much more sense.

**Comment by avikivity** [comment]
Updated subject for posterity.

**Comment by juliayakovlev** [comment]
Is this fix backported to 2.3?

**Comment by juliayakovlev** [comment]
@psarna  Is this fix backported to 2.3?

**Comment by psarna** [comment]
Looks like it isn't. @avikivity, can we backport this bugfix?
